### ISSUE TITLE
Error "Tried to create an ACL with an invalid permission type."

### ISSUE THREAD

**Issue by yog27ray** [parse-community/parse-server#8473]
### New Issue Checklist

- [X] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [X] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [X] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [X] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
I am updating the parse server from "5.2.8" to "5.4.2", and getting the below error
```
TypeError: Tried to create an ACL with an invalid permission type.
    at new ParseACL (/project/node_modules/parse/lib/node/ParseACL.js:70:21)
    at ParseRole._handleSaveResponse (/project/node_modules/parse/lib/node/ParseObject.js:614:25)
    at /project/node_modules/parse/lib/node/ParseObject.js:3183:22
    at processTicksAndRejections (node:internal/process/task_queues:95:5) 
```
Whenever I got the above error the arg1 parameter in the ParseACL.js constructor is called with the below value.
```
{ __op: 'Delete' }
```

I noticed when starting from the fresh empty database. I need to start the server multiple times to start the server as before starting the server it creates some roles. Every attempt create a few more entries in the Role table, Once all roles are created Server starts.

I also tried resetting  acl in before save of Role ```request.object.setACL(new Parse.ACL());``` but still end up with same error.

Server
- Parse Server version: `5.4.2`
- Operating system:  Ubuntu
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Docker`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `4.2.3`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Local`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `JavaScript`
- SDK version: `3.4.2`

**Comment by dblythy** [comment]
This might be related to #8386. It would help if you could provide some of the code used to trigger this error (such as Cloud triggers as well), or open a PR with a failing test 😊

**Comment by yog27ray** [comment]
@dblythy  I have created a PR with a failing test case.

**Comment by mtrezza** [comment]
@yog27ray Thanks for [confirming](https://github.com/parse-community/parse-server/pull/8474#issuecomment-1463969936) that this issue only occurs in [5.x](https://github.com/parse-community/parse-server/pull/8474) and not in [6.x](https://github.com/parse-community/parse-server/pull/8476). Fixing this is up to the community as we only fix security vulnerabilities in LTS versions. If this is not picked up by the community we recommend to upgrade to 6.x.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.3](https://github.com/parse-community/parse-server/releases/tag/5.4.3)
